File: snippets/csharp/ClosedChannel.cs
using Breez.Sdk;

public class ClosedChannelSnippets
{
    public void PrepareRedeemOnchainFunds(BlockingBreezServices sdk, uint feeRate) 
    {
        // ANCHOR: prepare-redeem-onchain-funds  
        var destinationAddress = "bc1..";
        var satPerVbyte = feeRate; 
        try 
        {
            var prepareRedeemOnchainFundsResp = sdk.PrepareRedeemOnchainFunds(
                new PrepareRedeemOnchainFundsRequest(
                    destinationAddress,
                    satPerVbyte
                ));
        }
        catch (Exception)
        {
            // Handle error
        }
         // ANCHOR_END: prepare-redeem-onchain-funds
    }

    public void RedeemOnchainFunds(BlockingBreezServices sdk, uint satPerVbyte, string toAddress) 
    {
        // ANCHOR: redeem-onchain-funds  
        try 
        {
            var redeemOnchainFundsResp = sdk.RedeemOnchainFunds(
                new RedeemOnchainFundsRequest(
                    toAddress,
                    satPerVbyte
                ));
        }
        catch (Exception)
        {
            // Handle error
        }
         // ANCHOR_END: redeem-onchain-funds
    }

}

File: snippets/rust/src/closed_channel.rs
use std::sync::Arc;

use anyhow::Result;
use breez_sdk_core::*;

async fn prepare_redeem_onchain_funds(sdk: Arc<BreezServices>, fee_rate: u32) -> Result<()> {
    // ANCHOR: prepare-redeem-onchain-funds
    let sat_per_vbyte = fee_rate;
    let destination_address = String::from("bc1..");
    sdk.prepare_redeem_onchain_funds(PrepareRedeemOnchainFundsRequest {
        sat_per_vbyte,
        to_address: destination_address,
    })
    .await?;
    // ANCHOR_END: prepare-redeem-onchain-funds
    Ok(())
}

async fn redeem_onchain_funds(
    sdk: Arc<BreezServices>,
    sat_per_vbyte: u32,
    to_address: String,
) -> Result<()> {
    // ANCHOR: redeem-onchain-funds
    sdk.redeem_onchain_funds(RedeemOnchainFundsRequest {
        sat_per_vbyte,
        to_address,
    })
    .await?;
    // ANCHOR_END: redeem-onchain-funds
    Ok(())
}
